### ISSUE TITLE
Is there any way to add "DataLabels" like Highcharts. That is, to provide above/next to the bar/column the value of the y axis? 

### ISSUE THREAD

**Issue by rousbound** [cetz-package/cetz-plot#2]
I mean something like this:
![highchart](https://github.com/johannes-wolf/cetz/assets/40251117/50900a2c-46ac-4519-b83f-341a95b951fc)

**Comment by wlievens** [comment]
Is this planned to be merged into cetz-plot soon? I don't see a PR for it in this project.

**Comment by johannes-wolf** [comment]
The cetz-plot repo moved, therefore this PR is in the wrong repo.

**Comment by johannes-wolf** [comment]
Original PR is here: https://github.com/cetz-package/cetz/pull/516

**Comment by johannes-wolf** [comment]
It is not easy right now to implement this in a good way, because of how plot scaling works.

**Comment by ramkumarkb** [comment]
Would it be possible to have a `basic version` of data-labels for bar charts - i.e. w/o scaling (users usually preview and can manually adjust?) ? Thank you,

**Comment by johannes-wolf** [comment]
> Would it be possible to have a `basic version` of data-labels for bar charts - i.e. w/o scaling (users usually preview and can manually adjust?) ? Thank you,

Yes. I'll try to add something after refactoring the plot environment to make sub-plots possible.

**Comment by LevitatingBusinessMan** [comment]
Is there currently any quick way to get this to work? Does the old PR still merge cleanly?

**Comment by LevitatingBusinessMan** [comment]
The code still appears to work although the placement of the labels seems a bit off.

**Comment by johannes-wolf** [comment]
With cetz:0.4.0 a better implementation is possible, but cetz-plot needs some rework to support than (custom coordinate systems). This is a feature I am planning to implement.

**Comment by johannes-wolf** [comment]
With this branch: 
https://github.com/cetz-package/cetz-plot/pull/135

And this code:
```
#let data = (
  (0, 4),
  (1, 13),
  (2, 12),
  (3, 3),
  (4, 7),
  (5, 9),
)

#cetz.canvas({
  import cetz.draw: *
  import cetz-plot: *

  plot.plot(size: (4,4), x-tick-step: none, y-tick-step: none, {
    for (x, y) in data {
      plot.annotate({
        content((plot: (x, y)), [Test], anchor: "north", padding: 0.1em)
      })
    }

    plot.add-bar(data,
      x-key: 0,
      y-key: 1)
  })
})
```

You can place labels:
![Image](https://github.com/user-attachments/assets/91eab31a-3bad-4e0a-9e2b-fc24046e62a3)

**Comment by johannes-wolf** [comment]
> Very interesting. Is this going to be released soon? I really need it in a report.

Sorry, but I do not think I will have the time to release this soon. The PR is a hack that I do not want to release this way. I want to do a rewrite of how cetz-plot works internally. In the example, you can see, that the domain calculation of the axes is not correct, because of the labels.

Is using a local checkout a problem?

**Comment by wlievens** [comment]
I could use a local checkout, yes. It's part of an automated flow in a python module so no reason why that would not work. Thanks for the tip.

**Comment by LevitatingBusinessMan** [comment]
I applied the following patch for `src/chart/barchart.typ` which works for my needs.

@johannes-wolf I am incredibly thankful that you could cook something up this quickly, this is exactly what I was hoping to be possible.

```PATCH
diff --git a/src/chart/barchart.typ b/src/chart/barchart.typ
index 5a10ed4..571efd8 100644
--- a/src/chart/barchart.typ
+++ b/src/chart/barchart.typ
@@ -73,6 +73,8 @@
               x-format: auto,
               y-label: none,
               labels: none,
+              draw-val: false,
+              draw-val-padding: 0.2em,
               ..plot-args
               ) = {
   assert(type(label-key) in (int, str))
@@ -131,6 +133,14 @@
               plot-style: bar-style,
               ..plot-args,
     {
+      if draw-val {
+        for (i, xs, _) in data {
+          plot.annotate({
+            draw.content((plot: (xs.at(0), i)), str(xs.at(0)), anchor: "west", padding: draw-val-padding)
+          })
+        }
+      }
+
       plot.add-bar(data,
         x-key: 0,
         y-key: 1,
```

**Comment by johannes-wolf** [comment]
Link: #170
